### PR TITLE
Blog post gui challenge split button

### LINKED ISSUE #5933: [A11y][CSS] Building a split button component [9/1/2021]
* [Draft](https://docs.google.com/document/d/1SSjKaVF1l4GpLtrbmjjN37O2f1QGdr-9QZ5X3vf9zOY)

### PR DESCRIPTION
**Pull request by argyleink**
Fixes #5933

ToDo:
- [x] final reviews
- [x] add proper youtube embed key
- [x] add section on RTL?

### CODE REVIEW

File: src/site/content/en/building-a-split-button-component/index.md [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+### Parts
+
+Let's break down the essential parts of a split button before discussing their
+overall orchestration and final user experience.
+[VisBug](https://github.com/GoogleChromeLabs/ProjectVisBug)'s accessibility
+inspect tool is used here to help show a macro view of the component, surfacing
+aspects of the HTML, style and accessibility for each major part.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/URCIaZV3gyMvTiMjRYmK.png",
+alt="ALT_TEXT_HERE", width="800", height="581", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:66]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/URCIaZV3gyMvTiMjRYmK.png",
alt="The HTML elements that make up the split button.", width="800", height="581", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+
+#### Top level split button container
+
+The highest level component is an inline flexbox, with a class of
+`gui-split-button`, containing the [primary action](#the-primary-action-button)
+and the [`.gui-popup-button`](#the-popup-toggle-button).
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/SLjXwBPvsrEXBb1vcm6L.png",
+alt="ALT_TEXT_HERE", width="800", height="581", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:75]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/SLjXwBPvsrEXBb1vcm6L.png",
alt="The gui-split-button class inspected and showing the CSS properties used in this class.", width="800", height="581", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+
+The initially visible and focusable `<button>` fits within the container with
+two matching corner shapes for
+[focus](https://web.dev/learn/css/pseudo-classes/#:focus-:focus-within-and-:focus-visible),
+[hover](https://web.dev/learn/css/pseudo-classes/#:hover) and
+[active](https://web.dev/learn/css/pseudo-classes/#:active) interactions to
+appear contained within `.gui-split-button`.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/69gvxF4AfQoNki4pBtVb.png",
+alt="ALT_TEXT_HERE", width="800", height="644", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:87]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/69gvxF4AfQoNki4pBtVb.png",
alt="The inspector showing the CSS rules for <button>.", width="800", height="644", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+
+#### The popup toggle button
+
+The "popup button" support element is for activating and alluding to the list of
+secondary buttons. Notice it's not a `<button>` and it's not focusable. However,
+it is the positioning anchor for [`.gui-popup`](#the-popup-card) and host for `:focus-within` used
+to present the popup.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/p2R7pHvID9XR9JoHHzth.png",
+alt="ALT_TEXT_HERE", width="800", height="490", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:97]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/p2R7pHvID9XR9JoHHzth.png",
alt="The inspector showing the CSS rules for the class gui-popup-button.", width="800", height="490", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 5]
```diff
@@ -0,0 +1,768 @@
...
+
+#### The popup card
+
+This is a floating card child to its anchor
+[`.gui-popup-button`](#the-popup-toggle-button), positioned absolute and
+semantically wrapping the button list. 
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/LSeLIFhe7yuG9QoR1VM9.png",
+alt="ALT_TEXT_HERE", width="800", height="629", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:106]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/LSeLIFhe7yuG9QoR1VM9.png",
alt="The inspector showing the CSS rules for the class gui-popup", width="800", height="629", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 6]
```diff
@@ -0,0 +1,768 @@
...
+
+#### The secondary action(s)
+
+A focusable `<button>` with a slightly smaller font size than the [primary
+action button](#the-primary-action-button) features an icon and a complimentary
+style to the primary button.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/5GBSno8pMaNVRZJT7yxd.png",
+alt="ALT_TEXT_HERE", width="800", height="679", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:115]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/5GBSno8pMaNVRZJT7yxd.png",
alt="The inspector showing the CSS rules for <button>.", width="800", height="679", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 7]
```diff
@@ -0,0 +1,768 @@
...
+  touch-action: manipulation;
+  user-select: none;
+  -webkit-tap-highlight-color: transparent;
+}
+```
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/a55qj9gU1qmoTlnbQvDe.png",
+alt="ALT_TEXT_HERE", width="800", height="241", class="w-screenshot" %}
+
+#### Styling the `<button>`s
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:286]:
```suggestion
#### The `<button>` styling
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 8]
```diff
@@ -0,0 +1,768 @@
...
+  fill: var(--ontheme);
+
+  touch-action: manipulation;
+  user-select: none;
+  -webkit-tap-highlight-color: transparent;
+}
+```
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/a55qj9gU1qmoTlnbQvDe.png",
+alt="ALT_TEXT_HERE", width="800", height="241", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:284]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/a55qj9gU1qmoTlnbQvDe.png",
alt="The split button.", width="800", height="241", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 9]
```diff
@@ -0,0 +1,768 @@
...
+
+#### Styling the `<button>`s
+
+Buttons are very good at disguising how much code it takes. You might need to
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:288]:
```suggestion
Buttons are very good at disguising how much code is required. You might need to
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 10]
```diff
@@ -0,0 +1,768 @@
...
+%}
+
+#### Styling the popup button
+
+A `4ch` flexbox for centering an icon and anchoring a popup button list. Like
+the primary button, it is transparent until otherwise hovered or interacted
+with, and stretched to fill.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/fiSlBJyM8x52b5msQBUY.png",
+alt="ALT_TEXT_HERE", width="800", height="241", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:462]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/fiSlBJyM8x52b5msQBUY.png",
alt="The arrow part of the split button used to trigger the popup.", width="800", height="241", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 11]
```diff
@@ -0,0 +1,768 @@
...
+#### Styling the popup
+
+The `.gui-popup` element is a floating card button list using custom properties
+and relative units to be subtly smaller, interactively matched with the primary
+button, and on brand with its use of color. Notice the icons have less contrast,
+are thinner, and the shadow has a hint of brand blue to it. Like with buttons,
+strong UI/UX is a result of these little details stacking up.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/D7RWfwJW25G3kuzPSnlz.png",
+alt="ALT_TEXT_HERE", width="448", height="422", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:570]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/D7RWfwJW25G3kuzPSnlz.png",
alt="A floating card element.", width="448", height="422", class="w-screenshot" %}
```

Comment (rachelandrew) [reply src/site/content/en/building-a-split-button-component/index.md:570]: Is this image right? Above you are talking about the icons, this image is just the card.

File: src/site/content/en/building-a-split-button-component/index.md [hunk 12]
```diff
@@ -0,0 +1,768 @@
...
+    0 16px 16px -2px hsl(var(--shadow) / calc(var(--shadow-strength) + 20%))
+  ;
+}
+```
+
+The icons and buttons are given brand colors to style nicely within each dark
+and light themed card:
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/ddCiS1PqDtjDRESUG4U1.png",
+alt="ALT_TEXT_HERE", width="452", height="422", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:611]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/ddCiS1PqDtjDRESUG4U1.png",
alt="Links and icons for checkout, Quick Pay, and Save for later.", width="452", height="422", class="w-screenshot" %}
```

File: src/site/content/en/building-a-split-button-component/index.md [hunk 13]
```diff
@@ -0,0 +1,768 @@
...
+    width: 100%;
+  }
+}
+```
+
+The dark theme popup has text and icon shadow additions, plus a slightly more
+intense box shadow:
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/4vOdW4OD1i4uNRQdTWH5.png",
+alt="ALT_TEXT_HERE", width="434", height="412", class="w-screenshot" %}
```
Comment (rachelandrew) [inline src/site/content/en/building-a-split-button-component/index.md:637]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/4vOdW4OD1i4uNRQdTWH5.png",
alt="The popup in the dark theme.", width="434", height="412", class="w-screenshot" %}
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
Mostly alt text!

File: src/site/content/en/building-a-split-button-component/index.md (hunk 11 above)
Comment (argyleink) [reply src/site/content/en/building-a-split-button-component/index.md:570]: the image is right, but i agree, the flow feels off. I can move the svg styles to after the popup is done, it's def refinement work that's a bit early in the post. i'll make a commit that order this the svg and popup better thanks!

**Comment by argyleink** [comment]
shooting to merge / release 9/1 👍🏻

**Review by rachelandrew — APPROVED** [review]

**Comment by argyleink** [comment]
correction: shooting to release 9/8, it's a week early

**Pull request merged**